packages/block-editor-utils/src/controls/Number.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { TextControl as NumberControl } from '@wordpress/components';
+import { ControlProps } from '../types/index.js';
+
+function Number<T extends Record<string, any>>({
+  config,
+  props,
+}: ControlProps<T>) {
+  const onChange = (newContent: string) => {
+    props.setAttributes({ [config.name]: newContent });
```

Comment:
`newContent` here is string but we require the save them as numbers. You need to cast this to an integer.